pilot/pkg/networking/v1alpha3/listener.go

```diff
@@ -41,7 +53,40 @@ func BuildListeners(env model.Environment, node model.Proxy) ([]*xdsapi.Listener
 		// TODO: add listeners for other protocols too
 		return buildGatewayHTTPListeners(env.Mesh, env.IstioConfigStore, node)
 	case model.Ingress:
-		// TODO : Need v1alpha3 equivalent of buildIngressGateway
+		services, err := env.Services()
+		if err != nil {
+			return nil, err
+		}
+		var svc *model.Service
+		for _, s := range services {
+			if strings.HasPrefix(s.Hostname, istioIngress) {
+				svc = s
+				break
+			}
+		}
+		insts := make([]*model.ServiceInstance, 0, 1)
+		if svc != nil {
+			insts = append(insts, &model.ServiceInstance{Service: svc})
+		}
+		return buildIngressListeners(env.Mesh, insts, env.ServiceDiscovery, env.IstioConfigStore, node), nil
 	}
 	return nil, nil
 }
+
+func newHTTPListener(ip string, port int, name string, config *types.Struct) *xdsapi.Listener {
+	return &xdsapi.Listener{
```

Comment:
where is this even being used?

Reply:
Not sure - merged your PR and a lot of what I added is gone.